src/utils/vcs.rs

```diff
@@ -249,6 +249,54 @@ pub fn git_repo_head_ref(repo: &git2::Repository) -> Result<String> {
     }
 }
 
+pub fn git_repo_base_ref(repo: &git2::Repository, remote_name: &str) -> Result<Option<String>> {
+    // Get the current HEAD commit
+    let head_commit = repo.head()?.peel_to_commit()?;
+
+    // Try to find the remote tracking branch
+    let remote_branch_name = format!("refs/remotes/{remote_name}/HEAD");
+    let remote_ref = match repo.find_reference(&remote_branch_name) {
+        Ok(r) => r,
+        Err(_) => {
+            // If remote/HEAD doesn't exist, try to find the default branch
+            // First try common default branch names
+            for branch in &["main", "master", "develop"] {
+                let remote_branch = format!("refs/remotes/{remote_name}/{branch}");
+                if let Ok(r) = repo.find_reference(&remote_branch) {
+                    return find_merge_base_ref(repo, &head_commit, &r);
+                }
+            }
+            bail!("Could not find remote tracking branch for {}", remote_name);
+        }
+    };
```

Comment:
You can greatly simplify this as follows:

```suggestion
    let remote_ref = repo.find_reference(&remote_branch_name).or_else(|_| {
        // If remote/HEAD doesn't exist, try to find the default branch
        // First try common default branch names
        ["main", "master", "develop"]
            .iter()
            .map(|branch| format!("refs/remotes/{remote_name}/{branch}"))
            .filter_map(|remote_branch| repo.find_reference(&remote_branch).ok())
            .next()
            .ok_or_else(|| {
                anyhow::anyhow!("Could not find remote tracking branch for {}", remote_name)
            })
    })?;
```

As a general tip, whenever you find yourself writing a lot of branching code in Rust (e.g. loops and match statements), there is often, but not always, a simpler way to do it with method chaining